store/src/java/com/zimbra/cs/service/UserServlet.java

```diff
@@ -278,7 +279,11 @@ private void sendError(UserServletContext ctxt, HttpServletRequest req, HttpServ
             && (req.getMethod().equalsIgnoreCase("POST") || req.getMethod().equalsIgnoreCase("PUT"))) {
             resp.sendError(HttpServletResponse.SC_UNAUTHORIZED, L10nUtil.getMessage(MsgKey.errMustAuthenticate, req));
         } else {
-            resp.sendError(HttpServletResponse.SC_NOT_FOUND, message);
+            if (ctxt.getAuthAccount() instanceof GuestAccount && ctxt.basicAuthAllowed() ) {
+                resp.sendError(HttpServletResponse.SC_UNAUTHORIZED, L10nUtil.getMessage(MsgKey.errMustAuthenticate, req));
+            } else {
+                resp.sendError(HttpServletResponse.SC_NOT_FOUND, message);
```

Comment:
Doesn't this still allow account enumeration?

Reply:
Yea at first glance it appears we can still send different responses depending on if the account exists or not.  I didn't see any notes in the ticket specifying exactly what the desired response should be (irregardless of account existence).

Reply:
Actually this is to specifically handle the case with basic auth where account does not exist or it failed validation. It is this case that has been mentioned in the report.  Also for the case whether authentication succeeds but  mail item cannot be found , it still makes sense to return a 404 with message "item not found"

Reply:
Hmmm.  This seems to work based on smoke and mirrors, if you understand all the code paths to how you got here then this is how it should behave and there aren't even any comments explaining the assumptions at that point.  I don't have a warm fuzzy feeling about that.  Perhaps a generic `sendError` handler like this is not appropriate?

Reply:
I cannot claim to know all the code pathways, but this is the solution  I narrowed down after trying various scenarios and debugging to see what pathways are followed.